docs/proposals/image-updater-first-class-support.md

```diff
@@ -0,0 +1,207 @@
+title: First class status for Argo CD Image Updater
+authors:
+  - "@jaideepr97" # Authors' github accounts here.
+sponsors:
+  - @jaanfis
+  - @wtam2018        # List all interested parties here.
+reviewers:
+  - "@alexmt"
+  - TBD
+approvers:
+  - "@alexmt"
+  - TBD
+
+creation-date: 2022-12-01
+last-updated: 2022-12-01
+---
+
+# Provide first class status to Image Updater within Argo CD
+
+Argo CD Image Updater is a popular project which solves an important gap within the CI/CD workflow by providing the ability to update application workload images automatically. It is designed around Argo CD itself and naturally compliments it. This proposal makes a case for upgrading it to first class citizen status within Argo CD and the changes that would involve to the Application CRD.
+
+
+## Summary
+
+Argo CD users would benefit from having image updater included as a first class workload. Using Annotations to configure image updates is not the preferred option, hence this proposal suggests adding new fields to the Application CR spec to store image updater configuration fields, as well as new fields in the Application status to store the state of image updates, enabling new features in the image update in the future, which can all be leveraged by users of Argo CD. 
+
+
+## Motivation
+
+Argo CD Image Updater is a popular project which solves an important gap within the CI/CD workflow by providing the ability to update application workload images automatically (subject to appropriate constraints). It is designed around Argo CD itself and naturally compliments it. The configuration required for it revolves almost entirely around the Application resource itself. Currently the image updater uses annotations on the application resource to store configuration, which is not the preferred way to do this. It also doesn't store any state, and is, as such, unable to provide some key features that are discussed in further sections. Consolidating this configuration into the Application CR as first class fields in the spec and status is a natural next step in the Image Updater's journey. 
+
+### Goals
+
+- Propose updated Application CR fields to include configuration scoped for image updates
+- Make a case to merge image updater controller code into core Argo CD as its own controller 
+- outline future enhancements planned for image updater and it's growth within Argo CD 
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+### Use cases
+
+#### Use case 1:
+As a User of Argo CD, I would like to be able to have my application images updated automatically within expressed constraints when there is a new version available, natively within Argo CD. 
+
+#### Use case 2:
+As a user of Argo CD and Argo CD Image Updater, I would like to be able to express advanced Image update configurations through custom resource fields in my application
+ 
+#### Use case 3:
+As a user of Argo CD and Argo CD Image Updater, I would like to have image-updater honor rollbacks to my applications
```

Comment:
I suppose it would be, but would require a completely separate effort to introduce state in image updater

I just added it here as a supplementary use case to support the other use cases and help make it more compelling
can remove it if you think it is not providing any value